### ISSUE TITLE
Tinygrad - not supported op

### ISSUE THREAD

**Issue by janchk** [aifoundry-org/ETARS#27]
While trying to run tinygrad's OnnxRunner (main branch) stumbled across `op='NonZero' is not supported` error in vision part of smolVLA module. 
This module -> https://huggingface.co/ainekko/smolvla_libero_onnx/blob/main/smolvlm_vision.onnx
Need to locate this op in graph and decide what to do next.
Either extend Tinygrad or do some hack over existing onnx graph to remove this op.

*FYI*
List of supported onnx ops in tinygrad

`
['Neg', 'Reciprocal', 'Pow', 'Sqrt', 'Sign', 'Abs', 'Exp', 'Log', 'Mish', 'Sin', 'Cos', 'Tan', 'Asin', 'Acos', 'Atan', 'Relu', 'Sigmoid', 'MatMul', 'Floor', 'Ceil', 'IsNaN', 'Softplus', 'HardSwish', 'Where', 'Mul', 'Sinh', 'Cosh', 'Tanh', 'Softsign', 'Asinh', 'Acosh', 'Atanh', 'Elu', 'Celu', 'Selu', 'Round', 'Erf', 'If', 'Identity', 'Constant', 'Range', 'ImageDecoder', 'EyeLike', 'OptionalHasElement', 'OptionalGetElement', 'ConstantOfShape', 'Size', 'Shape', 'Not', 'Clip', 'IsInf', 'HardSigmoid', 'BiasGelu', 'FastGelu', 'PRelu', 'LeakyRelu', 'ThresholdedRelu', 'LogSoftmax', 'Binarizer', 'Add', 'Sub', 'Div', 'Less', 'LessOrEqual', 'Greater', 'GreaterOrEqual', 'Equal', 'And', 'Or', 'Xor', 'BitwiseAnd', 'BitwiseOr', 'BitwiseXor', 'BitwiseNot', 'Mod', 'Cast', 'CastLike', 'Max', 'Min', 'Mean', 'ReduceMax', 'ReduceMin', 'ReduceMean', 'ReduceSumSquare', 'ReduceProd', 'ReduceL1', 'ReduceL2', 'ReduceLogSum', 'ReduceLogSumExp', 'ArgMin', 'Reshape', 'Flatten', 'Expand', 'Shrink', 'Transpose', 'Squeeze', 'Unsqueeze', 'Tile', 'Concat', 'Slice', 'Split', 'Pad', 'CenterCropPad', 'AveragePool', 'MaxPool', 'ConvTranspose', 'MaxUnpool', 'GlobalMaxPool', 'Gemm', 'Einsum', 'CumSum', 'Trilu', 'Upsample', 'TopK', 'BatchNormalization', 'InstanceNormalization', 'LayerNormalization', 'SkipLayerNormalization', 'EmbedLayerNormalization', 'MeanVarianceNormalization', 'OneHot', 'DepthToSpace', 'SpaceToDepth', 'LRN', 'NegativeLogLikelihoodLoss', 'SoftmaxCrossEntropyLoss', 'AffineGrid', 'RMSNormalization', 'RotaryEmbedding', 'ArrayFeatureExtractor', 'Gather', 'Scatter', 'GatherND', 'ScatterND', 'GatherElements', 'Compress', 'QuantizeLinear', 'DynamicQuantizeLinear', 'DequantizeLinear', 'QLinearConv', 'QLinearMatMul', 'QLinearAdd', 'QLinearMul', 'QLinearGlobalAveragePool', 'ConvInteger', 'MatMulInteger', 'Adagrad', 'Adam', 'Momentum', 'Gradient', 'ArgMax', 'Conv', 'Gelu', 'GlobalAveragePool', 'GroupNormalization', 'ReduceSum', 'Resize', 'ScatterElements', 'Sum', 'Softmax', 'Dropout', 'Attention']
`

**Comment by janchk** [comment]
Got it.
`/vision_model/embeddings/NonZero`

<img width="321" height="423" alt="Image" src="https://github.com/user-attachments/assets/ce1923b0-f850-4c91-8a48-b835c7f7f48a" />

The op definition from onnx docs.
https://onnx.ai/onnx/operators/onnx__NonZero.html

Oh wow
     
*This version of the operator has been available since version 13.*

And there is also a version from the 9th revision of onnx.

**Returns the indices of the elements that are non-zero.**

I'm afraid it's important part of embedding calculation.

**Comment by jerenkrantz** [comment]
From a cursory look, it might be problematic for tinygrad to easily add `NonZero` as an operator since it wants all of the shapes defined up-front rather than dynamically (a la ETGlow EP actually).

A few potentially helpful links:

Due to similar requirements for upfront knowledge of shape sizes, TensorRT didn't support `NonZero` for a long time either (though does now): https://github.com/onnx/onnx-tensorrt/issues/401

A story of `NonZero` support:
https://medium.com/@tomdol/generating-tensor-indices-a-story-of-nonzero-1-c07c9bf00def
https://medium.com/@tomdol/a-story-of-nonzero-2-390e2220cd86

**Comment by janchk** [comment]
Implemented and tested - https://github.com/janchk/tinygrad/commit/549e8e411de460117fdbaa2abe769e3caaeab9cf
Opened PR to the upstream - https://github.com/tinygrad/tinygrad/pull/13211

**Comment by janchk** [comment]
PR merged.
